### PR TITLE
PP-5354 Add DAO for govukpay_event TABLE

### PR DESCRIPTION
**Pull request by stephencdaly**
Add DAO with method for persisting a GovUkPayEvent.

### CODE REVIEW

File: src/main/java/uk/gov/pay/directdebit/events/model/GovUkPayEvent.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        this.eventDate = eventDate;
+        this.resourceType = resourceType;
+        this.eventType = eventType;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public Long getMandateId() {
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/directdebit/events/model/GovUkPayEvent.java]: Return an `Optional` for this?

File: src/main/java/uk/gov/pay/directdebit/events/model/GovUkPayEvent.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    private final ResourceType resourceType;
+    private final String eventType;
+
+    public GovUkPayEvent(Long id, Long mandateId, Long paymentId, ZonedDateTime eventDate, ResourceType resourceType, String eventType) {
+        this.id = id;
+        this.mandateId = mandateId;
+        this.paymentId = paymentId;
+        this.eventDate = eventDate;
+        this.resourceType = resourceType;
+        this.eventType = eventType;
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/directdebit/events/model/GovUkPayEvent.java]: Can we use `Objects.requireNonNull(…)` for the fields that should always be populated?

**Comment by stephencdaly** [comment]
We don't have external IDs that I can see for any other event. What would we use it for?

File: src/main/java/uk/gov/pay/directdebit/events/model/GovUkPayEvent.java (hunk 1 above)
Comment (stephencdaly) [reply src/main/java/uk/gov/pay/directdebit/events/model/GovUkPayEvent.java]: done

File: src/main/java/uk/gov/pay/directdebit/events/model/GovUkPayEvent.java (hunk 2 above)
Comment (stephencdaly) [reply src/main/java/uk/gov/pay/directdebit/events/model/GovUkPayEvent.java]: done

**Comment by alexbishop1** [comment]
You’re right, probably nothing.

**Pull request merged**